crates/admin/src/cluster_controller/service.rs

```diff
@@ -288,17 +288,25 @@ where
         }
 
         for (partition_id, persisted_lsns) in persisted_lsns_per_partition.into_iter() {
-            // todo: Check that we haven't forgotten a replica which is not part of the cluster state, yet
-            let min_persisted_lsn = persisted_lsns.into_values().min().unwrap_or(Lsn::INVALID);
             let log_id = LogId::from(partition_id);
-            // trim point is before the oldest record
-            let current_trim_point = bifrost_admin.get_trim_point(log_id).await?;
 
-            if min_persisted_lsn >= current_trim_point + self.log_trim_threshold {
-                debug!(
+            // todo: Remove once Restate nodes can share partition processor snapshots
+            // only try to trim if we know about the persisted lsns of all known nodes; otherwise we
+            // risk that a node cannot fully replay the log; this assumes that no new nodes join the
+            // cluster after the first trimming has happened
+            if persisted_lsns.len() >= cluster_state.nodes.len() {
```

Comment:
If we assumed sticky scheduling of partitions to a subset of the nodes, then this would also work. At the moment, the PR assumes that partition processors can be scheduled on all available nodes. Moreover, it assumes that no new nodes will be added to the cluster once the first trimming happens.